src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -1736,15 +1736,15 @@ static int32_t guest_path_find_leaving_park(Peep* peep, uint8_t edges)
 static int32_t guest_path_find_park_entrance(Peep* peep, uint8_t edges)
 {
     // If entrance no longer exists, choose a new one
-    if ((peep->PeepFlags & PEEP_FLAGS_PARK_ENTRANCE_CHOSEN) && peep->ChosenParkEntrance >= gParkEntrances.size())
+    if ((peep->PeepFlags & PEEP_FLAGS_PARK_ENTRANCE_CHOSEN) && peep->ChosenParkEntrance.ToUnderlying() >= gParkEntrances.size())
     {
-        peep->ChosenParkEntrance = PARK_ENTRANCE_INDEX_NULL;
+        peep->ChosenParkEntrance = ParkEntranceIndex::Null;
         peep->PeepFlags &= ~(PEEP_FLAGS_PARK_ENTRANCE_CHOSEN);
     }
 
     if (!(peep->PeepFlags & PEEP_FLAGS_PARK_ENTRANCE_CHOSEN))
     {
-        uint8_t chosenEntrance = PARK_ENTRANCE_INDEX_NULL;
+        ParkEntranceIndex chosenEntrance = ParkEntranceIndex::Null;
```

Comment:
This is a good opportunity to use `auto`.

Reply:
The type has to be ParkEntranceIndex and not ParkEntranceIndex::ValueType to be able to use FromUnderlying in the loop.